### DISCUSSION TITLE
如何指定其他包管理器搜索？

### DISCUSSION THREAD

**Discussion by ShelpAm** [xmake-io/xmake#5493] [Q&A] [answered]
比如我想搜索vcpkg里的zli,我尝试了xrepo search vcpkg::zlib,但是它返回了:
```
The package names:
error: vcpkg not found!
error: execv(/usr/bin/xmake require --search vcpkg::zlib) failed(255)
```

我应该怎么做？

**Comment by star-hengxing** [comment]
xrepo search 没这个功能。直接用 add requires。

**Reply by waruqi** [reply]
> xrepo search 没这个功能。直接用 add requires。

有的，只是 vcpkg 没找到而已，配置下 VCPKG_ROOT 就行了。

**Reply by ShelpAm** [reply]
> > xrepo search 没这个功能。直接用 add requires。
> 
> 有的，只是 vcpkg 没找到而已，配置下 VCPKG_ROOT 就行了。

```
❯ VCPKG_ROOT=/bin xrepo search vcpkg::zlib
The package names:
```
这是不是没找到呀？没有这个库吗。。

**Comment by waruqi** [comment, marked as answer]
我这里可以，要么就是 vcpkg root 指定不对，要么就是你 vcpkg 安装有问题。。

你可以自己调下 https://github.com/xmake-io/xmake/blob/dev/xmake/modules/package/manager/vcpkg/search_package.lua 

```console
$ xrepo search vcpkg::zlib
The package names:
    vcpkg::zlib: 
      -> vcpkg::zlib-1.2.11#9: A compression library 
```

**Reply by ShelpAm** [reply]
请问怎么调啊？麻烦了... 有文档也可以。

**Reply by star-hengxing** [reply]
直接在源码里用 print 调试或 https://github.com/xmake-io/xmake/issues/4122
